This patch is based on the following commit, with the new Implies
files for POWER10, and the preconfigure changes removed.

commit d2ba3677da7a785556fcd708404d8e049b1c063b
Author: Tulio Magno Quites Machado Filho <tuliom@linux.ibm.com>
Date:   Wed Jun 24 18:04:41 2020 -0300

    powerpc: Add support for POWER10
    
    1. Add the directories to hold POWER10 files.
    
    2. Add support to select POWER10 libraries based on AT_PLATFORM.
    
    3. Let submachine=power10 be set automatically.


diff --git a/sysdeps/powerpc/dl-procinfo.h b/sysdeps/powerpc/dl-procinfo.h
index 3803379ab2303658..3558d6a83ca2a988 100644
--- a/sysdeps/powerpc/dl-procinfo.h
+++ b/sysdeps/powerpc/dl-procinfo.h
@@ -37,7 +37,7 @@
 #define HWCAP_IMPORTANT		(PPC_FEATURE_HAS_ALTIVEC \
 				+ PPC_FEATURE_HAS_DFP)
 
-#define _DL_PLATFORMS_COUNT	15
+#define _DL_PLATFORMS_COUNT	16
 
 #define _DL_FIRST_PLATFORM	32
 /* Mask to filter out platforms.  */
@@ -60,6 +60,7 @@
 #define PPC_PLATFORM_PPC476		12
 #define PPC_PLATFORM_POWER8		13
 #define PPC_PLATFORM_POWER9		14
+#define PPC_PLATFORM_POWER10		15
 
 static inline const char *
 __attribute__ ((unused))
@@ -91,6 +92,14 @@ _dl_string_platform (const char *str)
       str += 5;
       switch (*str)
 	{
+	case '1':
+	  if (str[1] == '0')
+	    {
+	      ret = _DL_FIRST_PLATFORM + PPC_PLATFORM_POWER10;
+	    }
+	  else
+	    return -1;
+	  break;
 	case '4':
 	  ret = _DL_FIRST_PLATFORM + PPC_PLATFORM_POWER4;
 	  break;
